### ISSUE TITLE
Rpc calls made by web3.js over websockets do not get a response.

### ISSUE THREAD

**Issue by RobbeVerhelst** [emeraldpay/dshackle#163]
I found this issue when trying to benchmark different Hyperledger Besu nodes using Hyperledger Caliper. Our setup is x amount of non-validator nodes behind a dshackle loadbalancer. 

When running Caliper we saw that we didn't get any response from dshackle over websockets. When we ran Caliper directly on a besu node everything worked fine. After some further investigation we saw Caliper uses web3.js to make rpc calls. So to test this out we made a small web3 project and found that when web3 tries to send rpc calls over websockets to dshackle it does not get a reply. We do however see the rpc call coming into dshackle (in a grafana dashboard), but the web3 project does not get a reply. 

Somewhere in the websocket part of dshackle it is not able to send the response to the web3 project.

**Comment by splix** [comment]
What method do you call over websockets? It's possible that those methods are just not enabled by default

**Comment by RobbeVerhelst** [comment]
Unfortunately that's not possible, I call eth_blockNumber and am able to get the response from that using wscat and even ethers.js using ws. 
It's specifically web3.js that's not getting a response for some reason

**Comment by splix** [comment]
Do you mean that the a request for `eth_blockNumber` made with wscat works, but the _exactly same request_ with web3.js doesn't work?

**Comment by splix** [comment]
I mean, I'm sure there is some difference in the call that make one request working and another one not

**Comment by RobbeVerhelst** [comment]
Yes, I'm sending the same request using web3.js as I am with wscat, but also with ethers.js I get a response. Specifically web3.js is not getting the answer from dshackle, but dshackle does get the request and it is "successful" when checking the grafana dashboard. But on the web3 side I get nothing. So web3.js does something different then ethers.js or wscat.

**Comment by splix** [comment]
Can you trace the requests with Wireshark and provide me with the details?

**Comment by PDVJAM** [comment]
Any updates on this? We actually have the same issue with web3 and websockets.

**Comment by splix** [comment]
@PDVJAM can you provide with traffic traces between the dshackle and the upstreams? We need something to look at to find out what is the problem

**Comment by splix** [comment]
The latest version of the code in `master` supports logging of all requests, including requests to the upstreams. So if you are able try with a version from master that could provide enough details

**Comment by PDVJAM** [comment]
> The latest version of the code in `master` supports logging of all requests, including requests to the upstreams. So if you are able try with a version from master that could provide enough details

Is there a docker image already or must be build manually?

**Comment by brixial** [comment]
i guess this also has the same context?
#224
